src/rotation.zig

```diff
@@ -507,13 +591,88 @@ pub const RotationModes = struct {
 		pub const id: []const u8 = "stairs";
 		var modelIndex: u16 = 0;
 
-		fn subBlockMask(x: u1, y: u1, z: u1) u8 {
+		inline fn subBlockMask(x: u1, y: u1, z: u1) u8 {
 			return @as(u8, 1) << ((@as(u3, x)*2 + @as(u3, y))*2 + z);
 		}
-		fn hasSubBlock(stairData: u8, x: u1, y: u1, z: u1) bool {
+		inline fn hasSubBlock(stairData: u8, x: u1, y: u1, z: u1) bool {
 			return stairData & subBlockMask(x, y, z) == 0;
 		}
 
+		fn rotateX(data: u16) u16 {
+			comptime var rotateTable: [8][3]u1 = undefined;
+			comptime  for(0..2) |i| for(0..2) |j| for(0..2) |k| {
+				const sin: f32 = @sin(std.math.pi / 2.0);
+				const cos: f32 = @cos(std.math.pi / 2.0);
+				const x: f32 = (@as(f32, @floatFromInt(i)) - 0.5) * 2.0;
+				const y: f32 = (@as(f32, @floatFromInt(j)) - 0.5) * 2.0;
+				const z: f32 = (@as(f32, @floatFromInt(k)) - 0.5) * 2.0;
+				rotateTable[i*4 + j*2 + k] = .{
+					@intFromBool(x > 0),
+					@intFromBool(y*cos - z*sin > 0),
+					@intFromBool(y*sin + z*cos > 0),
+				};
+			};
+			var new: u8 = 0;
+			for(0..2) |i| for(0..2) |j| for(0..2) |k| {
+				const x: u1 = @truncate(i);
+				const y: u1 = @truncate(j);
+				const z: u1 = @truncate(k);
+				const xyz = rotateTable[i*4 + j*2 + k];
+				new |= subBlockMask(xyz[0], xyz[1], xyz[2]) * @intFromBool(hasSubBlock(@truncate(data), x, y, z));
```

Comment:
This part here is to avoid conditional branch, it is possible that this is excessive, maybe optimizer does that anyway, I didn't check that.

Reply:
Yes, most optimizers will just use a `cmov` instruction here. Using multiplication may actually make it harder for the optimizer to discover your intentions.

Reply:
Furthermore a branch might even be better, if we assume that most of the time stair blocks are fully filled.

Reply:
```suggestion
				if(hasSubBlock(@truncate(data), x, y, z)) {
					new |= subBlockMask(xyz[0], xyz[1], xyz[2]);
				}
```